fendermint/eth/api/src/apis/mod.rs

```diff
@@ -6,20 +6,66 @@
 
 use crate::HybridClient;
 use jsonrpc_v2::{MapRouter, ServerBuilder};
-use paste::paste;
+use lazy_static::lazy_static;
+use prometheus::{register_histogram_vec, HistogramVec};
 
 mod eth;
 mod net;
 mod web3;
 
-macro_rules! with_methods {
+// TODO - move this to a more appropriate place - perhaps in the metrics module?
+lazy_static! {
+    pub static ref RPC_METHOD_CALL_LATENCY_SECONDS: HistogramVec = register_histogram_vec!(
+        "rpc_method_call_duration_seconds",
+        "Histogram of RPC method call durations",
+        &["method"]
+    )
+    .unwrap();
+}
+
+// TODO - find a better way to generate these methods ideally with single macro or without macros.
+macro_rules! with_methods_one_arg {
```

Comment:
Hey @cryptoAtwill, I wonder if there is a better way instead of using two macros to register these methods. The issue is that some of the handlers have two arguments while others have one. Do you have any ideas, please? Any suggestions to simplify or unify the registration process would be highly appreciated.

Reply:
Seems compiles: https://github.com/consensus-shipyard/ipc/pull/1048